ktor-features/ktor-metrics-micrometer-prometheus/jvm/src/io/ktor/metrics/micrometer/prometheus/MicrometerMetricsPrometheus.kt

```diff
@@ -0,0 +1,149 @@
+            val metricSamplesToScrape = value.filteredMetricFamilySamples()
+
+            channel.bufferedWriter(contentType.charset() ?: Charsets.UTF_8).use {
+                TextFormat.write004(it, metricSamplesToScrape)
+            }
+        }
+    }
+
+    class Configuration : AbstractMicrometerMetrics.AbstractConfiguration<PrometheusMeterRegistry>() {
+
+        /**
+         * The meter registry
+         */
+        override var registry = PrometheusMeterRegistry(PrometheusConfig.DEFAULT)
+
+        override var distributionStatisticConfig: DistributionStatisticConfig = DistributionStatisticConfig.Builder()
+            .percentilesHistogram(true)
+            .build()
+
+        /**
+         * Defines if a metrics endpoint is installed under "/metrics" for all metrics registered at [registry].
+         * Set to false if you want to declare a custom metric endpoint instead.
+         */
+        var installDefaultEndpoint = true
+    }
+
+
+    companion object MetricsFeature :
+        AbstractMetricsFeature<PrometheusMeterRegistry, Configuration>(
+            ::MicrometerMetricsPrometheus,
+            MicrometerMetricsPrometheus::Configuration
+        ) {
+
+        override fun install(
+            pipeline: Application,
+            configure: Configuration.() -> Unit
+        ): AbstractMicrometerMetrics<PrometheusMeterRegistry, Configuration> {
+            val feature = super.install(pipeline, configure) as MicrometerMetricsPrometheus
+
+            pipeline.sendPipeline.intercept(ApplicationSendPipeline.Transform) { value ->
+                if (value is MetricsContent) {
+                    val response = feature.process(value)
+                    proceedWith(response)
+                }
+            }
+
+            if (feature.config.installDefaultEndpoint)
+                pipeline.routing {
+                    get("/metrics") {
+                        call.respondMetrics()
+                    }
+                }
+
+            return feature
+        }
+    }
+}
+
+
+/**
+ * the meter registry registered for this feature
+ */
+val ApplicationCall.meterRegistry
+    get() = application.feature(MicrometerMetricsPrometheus).config.registry
+
+/**
+ * responds with scraped metrics of this registry, filtered by the specified filter. The filter works after the names
+ * of the metrics have been "prometheusized" meaning that the names and tags
+ */
+suspend fun ApplicationCall.respondMetrics(
+    registry: PrometheusMeterRegistry = meterRegistry,
+    filter: Collector.MetricFamilySamples.() -> Boolean = { true }
+) =
+    respond(MetricsContent(registry, filter))
+
+
+internal class MetricsContent(
+    val registry: PrometheusMeterRegistry,
+    val filter: Collector.MetricFamilySamples.() -> Boolean
+) {
+    fun filteredMetricFamilySamples(): Enumeration<Collector.MetricFamilySamples> =
+        object : Enumeration<Collector.MetricFamilySamples> {
+
+            private val internalCollection =
```

Comment:
`registry.prometheusRegistry.metricFamilySamples().iterator().asSequence().filter(filter).iterator()` could replace a lot of the following code